Automattic/vip-go-mu-plugins
# Add VIP NotOptions Mitigation

Detect invalid values in the notoptions cache.

## Description

The notoptions cache should only ever include keys whose value is `true` (bool).
In cases where that value is corrupted (these circumstances not addressed here, this is solely mitigation), flush cache and alert the VIP team.

- only run in admin or CLI context
- only alert in production environments

Historical VIP discussion:

- https://wp.me/p9GBOq-2bh
- https://wp.me/p9GBOq-27U
- 
## Changelog Description

### Options: Add code to detect invalid values in the notoptions cache (Inactive)

Adds a new file with code that automatically detects corruption of the `notoptions` cache entry, clearing the cache and alerting when it is detected. This code is not yet activated - it will be required by a follow-up PR.

## Checklist

Please make sure the items below have been covered before requesting a review:

- n/a This change has relevant unit tests (if applicable).
- n/a This change has relevant documentation additions / updates (if applicable).

## Steps to Test

With the PR applied, purposefully contaminate notoptions:

```
$notoptions = wp_cache_get( 'notoptions', 'options' );
$notoptions['somekey'] = 'anystring';
wp_cache_set( 'notoptions', $notoptions, 'options' );
```

*must be a `production` environment to run alerting code, else alter that condition*

Then, any CLI command will meet conditions to run the check:

```
wp option get home
```
